src/test/compile-fail/const-fn-not-safe-for-const.rs

```diff
@@ -12,12 +12,12 @@
 
 #![feature(const_fn)]
 
-use std::mem::transmute;
+use std::mem::size_of_val;
 
 fn random() -> u32 { 0 }
 
 const fn sub(x: &u32) -> usize {
-    unsafe { transmute(x) } //~ ERROR E0015
```

Comment:
so... I have a bad feeling about these intrinsics. The fact that this change was necessary makes me wonder whether the intrinsics have become insta-stable (because e.g. [`std::mem::transmute`](https://doc.rust-lang.org/std/mem/fn.transmute.html) is the intrinsic directly, without a wrapper function).

Please add tests that make sure that each intrinsic's stable version does not become const fn on stable Rust.

It's probably best to add some tests that check whether the intrinsics work with feature gates, too.

Reply:
There should probably be a `#[const_unstable_...` attribute on the version of `transmute` that's marked as stable, and somehow have that checked in `qualify_consts`.